OpenRA.Game/Player.cs

```diff
@@ -52,8 +52,19 @@ static CountryInfo ChooseCountry(World world, string name)
 				.WithInterface<CountryInfo>().Where(c => c.Selectable)
 				.ToList();
 
-			return selectableCountries.FirstOrDefault(c => c.Race == name)
+			var selected = selectableCountries.FirstOrDefault(c => c.Race == name)
 				?? selectableCountries.Random(world.SharedRandom);
+
+			if (selected.RandomRace.Any())
```

Comment:
You could solve (most of) the problem Phrohdoh mentioned immediately above by changing this `if` to `while`. However, if the modder screws it up and creates a loop, you'd get an infinite loop here. On the plus side, you'd get support for multiple levels of random races.